examples/kv_addr_book/include/kv_addr_book.hpp

```diff
@@ -0,0 +1,153 @@
+
+   eosio::name get_account_name() const {
+      return this->account_name;
+   }
+   string get_first_name() const {
+      return get<1>(this->first_name);
+   }
+   string get_last_name() const {
+      return get<1>(this->last_name);
+   }
+   string get_street() const {
+      return get<1>(this->street_city_state_cntry);
+   }
+   string get_city() const {
+      return get<2>(this->street_city_state_cntry);
+   }
+   string get_state() const {
+      return get<3>(this->street_city_state_cntry);
+   }
+   string get_country() const {
+      return get<4>(this->street_city_state_cntry);
+   }
+   string get_personal_id() const {
+      return get<1>(this->personal_id);
+   }
+};
+
+// helper factory to easily build person objects
+struct person_factory {
+   static person get_person(
+      eosio::name account_name,
+      string first_name,
+      string last_name,
+      string street,
+      string city,
+      string state, 
+      string country,
+      string personal_id) {
+         return person {
+            .account_name = account_name,
+            .first_name = {account_name, first_name},
+            .last_name = {account_name, last_name},
+            .street_city_state_cntry = {account_name, street, city, state, country},
+            .personal_id = {account_name, personal_id},
+            .country_personal_id = {country, personal_id}
+         };
+      }
+};
+
+struct address_table : kv_table<person> {
+   // unique indexes definitions
+   // 1. they are defined for just one property of the kv_table parameter type (person)
+   // 2. unique indexes for multiple properties of the kv_table parameter type
+   //    are defined with the help of a pair or a tuple; a pair if the index has 
+   //    two properties or a tuple in case of more than two
+   index<name>   account_name_uidx {"accname"_n, &person::account_name};
+   index<pair<string, string>> country_personal_id_uidx {"cntrypersid"_n, &person::country_personal_id};
+   
+   // non-unique indexes definitions
+   // 1. non unique indexes need to be defined for at least two properties, 
+   // 2. the first one needs to be a property that stores unique values, because 
+   //    under the hood every index (non-unique or unique) is stored as an unique 
+   //    index, and by providing as the first property one that has unique values
+   //    it ensures the uniques of the values combined (including non-unique ones)
+   // 3. the rest of the properties are the ones wanted to be indexed non-uniquely
+   index<non_unique<eosio::name, string>> first_name_idx {
+      "firstname"_n, &person::first_name};
+   index<non_unique<eosio::name, string>> last_name_idx {
+      "lastname"_n, &person::last_name};
+   index<non_unique<eosio::name, string>> personal_id_idx {
+      "persid"_n, &person::personal_id};
+   index<non_unique<eosio::name, string, string, string, string>> address_idx {
+      "address"_n, &person::street_city_state_cntry};
```

Comment:
I'd like to see the KV_NAMED_INDEX macro demo'd for at least one of these indices. I imagine most people will use the macro and it would make the example look a little less intimidating by removing a lot of code.